File: 748.go
package leetcode

import (
	"unicode"
)

func shortestCompletingWord(licensePlate string, words []string) string {
	cnt := map[rune]int{}
	for _, r := range licensePlate {
		if unicode.IsLetter(r) {
			cnt[unicode.ToLower(r)]++
		}
	}
	var ans string
	ans_len := 1000
outer:
	for _, w := range words {
		if len(w) < ans_len {
			cntW := map[rune]int{}
			for _, r := range w {
				cntW[r]++
			}
			for key, val := range cnt {
				if val > cntW[key] {
					continue outer
				}
			}
			ans = w
			ans_len = len(ans)
		}
	}
	return ans
}

// func main() {
// 	a := "1s3 PSt"
// 	b := []string{"step", "steps"}
// 	fmt.Println(shortestCompletingWord(a, b))
// }

File: 748.py
from collections import Counter
from typing import List


class Solution:
    def shortestCompletingWord(self, licensePlate: str, words: List[str]) -> str:
        licensePlate = [c.lower() for c in licensePlate if c.isalpha()]
        C0 = Counter(licensePlate)
        ans = None
        ans_len = float('inf')
        for w in words:
            if len(w) < ans_len:
                if not C0 - Counter(w):
                    ans = w
                    ans_len = len(ans)
        return ans
